### PR TITLE
Support for the new Figma share URL

### LINKED ISSUE #239: Support for the updated Figma embed
### Affected design types

- [X] figma
- [X] figspec
- [ ] link
- [ ] image
- [ ] iframe

### Describe the bug

I don't know if this is a very recent change or not. I've just now starting using the figma integration with storybook. I know figma has been going through lots of changes regarding developer mode, making it an additional cost and all... Maybe this is new URL is part of those recent changes.

Right-clicking on a figma variant, select `Copy as` > `Copy link to selection`, it gives you a URL that starts with
`https://www.figma.com/design/...`

Entering this into the figma or figspec parameters, the design panel then displays an error with the URL and `is not a valid Figma URL`

It appears figma is still supporting the `file` path, it just forwards you to a `design` path in the end. So my workaround so far has been to use the Copy link to selection and simply swap out `design` with `file`.

Would be nice to update the regex figmaURLPattern in packages/storybook-addon-designs/src/manager/components/Figma.tsx to include `design`

### How to reproduce the bug?

1. Right-clicking on a figma variant, select `Copy as` > `Copy link to selection`. Note, the copied URL starts with
`https://www.figma.com/design/...`
2. Entering this URL into a figma or figspec parameters:url definition of a story
3. The design panel then displays an error with the URL and `is not a valid Figma URL`

### Expected behaviour

Should display figma without an error for a valid url

### Environment

- [X] Chrome / Chromium / Chromium based browsers
- [ ] Firefox
- [ ] macOS Safari
- [ ] iOS Safari
- [ ] Samsung Internet
- [ ] Other

### Affected versions

8.0.1

### Storybook versions

8.0.4

### PR DESCRIPTION
**Pull request by pocka**
close #239

Figma changed URL structure for the share URL from `"/file/..."` to `"/<anything>/..."`. This patch updates the URL checking regexp according to their updated [embed document](https://www.figma.com/developers/embed).

<details>
  <summary>📦 Published PR as canary version: <code>8.0.3--canary.240.540483c.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/addon-designs@8.0.3--canary.240.540483c.0
  # or 
  yarn add @storybook/addon-designs@8.0.3--canary.240.540483c.0
  ```
</details>

### CODE REVIEW

File: packages/storybook-addon-designs/src/manager/components/Figma.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { IFrame } from "./IFrame";
 import { FigmaConfig, IFrameConfigBase } from "../../config";
 
 export const figmaURLPattern =
-  /https:\/\/([w.-]+.)?figma.com\/(file|proto)\/([0-9a-zA-Z]{22,128})(?:\/.*)?$/;
+  /https:\/\/[\w\.-]+\.?figma.com\/([\w-]+)\/([0-9a-zA-Z]{22,128})(?:\/.*)?$/;
```
Comment (pocka) [inline packages/storybook-addon-designs/src/manager/components/Figma.tsx:10]: As Codacy annotates, the first `\.` is totally unnecessary. However, this is from Figma docs so I kept that as-is.

**Review by ghengeveld — APPROVED** [review]

File: packages/storybook-addon-designs/src/manager/components/Figma.tsx (hunk 1 above)
Comment (ghengeveld) [inline packages/storybook-addon-designs/src/manager/components/Figma.tsx:10]:
```suggestion
  /https:\/\/[\w.-]+\.?figma.com\/([\w-]+)\/([0-9a-zA-Z]{22,128})(?:\/.*)?$/;
```

Comment (ghengeveld) [reply packages/storybook-addon-designs/src/manager/components/Figma.tsx]: I ended up following Codacy's advice anyway. Easier to just keep it happy than to work around it.

**Comment by ed-ateixeira** [comment]
any updates when this should be merged?

**Pull request merged**